static/js/app.js

```diff
@@ -46,11 +51,14 @@ textArea.on('mouseup', function () {
       'endPosition': textArea[0].selectionEnd
     };
     if (range.startPosition < range.endPosition) {
-      console.log("Highlighted: " + range.startPosition + "-" + range.endPosition);
-      tagModel.addAnnotation(range);
+      let annotationCreated = tagModel.addAnnotation(range, tagModel.currentCategory);
```

Comment:
If we just need the annotation object and not the document its encompassed in, I think it would make more sense to just call the newAnnotation() function here. Having a method name that implies work being done that returns an object is a bit counter-intuitive. (addAnnotation vs. getAnnotation, for ex.)

Reply:
For the annotations list feature, we will need the document it is encompassed in to display the annotations for that particular document. This should stay as is.